contrail-provisioner/analyticsnodes/analyticsnodes.go

```diff
@@ -0,0 +1,180 @@
+package analyticsnodes
+
+import (
+	"log"
+	"os"
+
+	contrailTypes "github.com/Juniper/contrail-operator/contrail-provisioner/contrail-go-types"
+	"github.com/Juniper/contrail-operator/contrail-provisioner/contrailclient"
+)
+
+// AnalyticsNode struct defines Contrail Analytics node
+type AnalyticsNode struct {
+	IPAddress   string            `yaml:"ipAddress,omitempty"`
+	Hostname    string            `yaml:"hostname,omitempty"`
+	Annotations map[string]string `yaml:"annotations,omitempty"`
+}
+
+const analyticsNodeType string = "analytics-node"
+
+var analyticsInfoLog *log.Logger
+
+func init() {
+	analyticsInfoLog = log.New(os.Stdout, "analyticsnodes: ", log.LstdFlags)
+}
+
+// Create creates a AnalyticsNode instance
+func (c *AnalyticsNode) Create(nodeList []*AnalyticsNode, nodeName string, contrailClient contrailclient.ApiClient) error {
```

Comment:
Yes, I know it is weird. I didn't want to rewrite the whole code (because the same "pattern" is applied everywhere) in this PR, so I just copied it from main.go to a separate packages for now.
I cleaned it up a little bit in some previous PR for the vrouter nodes. Take a look for example at this method:
https://github.com/Juniper/contrail-operator/blob/1021cc8ccc46313c70a4a43584642ec766fefc9a/contrail-provisioner/vrouternodes/vrouter_nodes.go#L83

Reply:
Ok, refactored it to something similar to what you suggested. Let me know if it is fine